### ISSUE TITLE
ChromeCast.Device not visible

### ISSUE THREAD

**Issue by jbrepogmailcom** [SamDel/ChromeCast-Device#1]
Hello I launched ChromeCast.Device as instructed on Raspberry Pi on local WiFi and I would expect that I can see the device by Deezer on mobile device connected to same network. Deezer allows to stream to ChromeCast device. However the device is not found by Deezer mobile app.

**Comment by SamDel** [comment]
That doesn't work (yet).
For now the only app it works with is this one: https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer

**Comment by thomas-illiet** [comment]
@SamDeld do you have any idea to work with standard device ?
Actually i'm unable to work properly with the desktop audio streamer also :/

**Comment by SamDel** [comment]
> @SamDeld do you have any idea to work with standard device ?

Do you mean working together with the standard Google software (Home app, Chrome browser, etc.)? I don't know why this software isn't found by them, further reverse engineering is needed.

> Actually i'm unable to work properly with the desktop audio streamer also :/

Is it not found by desktop audio streamer? What's your setup? Are both devices connected to the same wifi network?

**Comment by thomas-illiet** [comment]
Yes reverse engineering is needed, i search with my little skill why the server is not connected to the client.
I solved my issue, because i trying to run your application on windows with systemcalls.cs update.

**Comment by SamDel** [comment]
> Yes reverse engineering is needed, i search with my little skill why the server is not connected to the client.

So far I only tried to make the MDNS response identical to the Chromecast response ([here in code](https://github.com/SamDel/ChromeCast-Device/blob/3cb5693aebfe1734c448cdfb6c5237b6c9045960/Source/ChromeCast.Device/Application/MdnsAdvertise.cs#L70)). Maybe something is missing there, or another API has to be implemented, or ...

> I solved my issue, because i trying to run your application on windows with systemcalls.cs update.

Nice! If you post the code we can create a multiplatform version!?

**Comment by thomas-illiet** [comment]
My target project is creating a proxy/relay for youtube video to remove ads
Client device -> Relay -> download video -> cast video from mp4 file to the official chromecast.

I search solution have virtual chromecast to do this :)

**Comment by thomas-illiet** [comment]
I found following project: https://github.com/tristanpenman/node-castv2
this fork is I find it very interesting i think

**Comment by SamDel** [comment]
> I found following project: https://github.com/tristanpenman/node-castv2
> this fork is I find it very interesting i think

Thanks, that might be interesting. I'll look into it later.
The description of this project says:

> ... The module provides both a Client and a Server implementation of the low-level protocol. **The server is (sadly) pretty useless because device authentication gets in the way for now (and maybe for good)**. ...

so I don't think it's a fully working example, but it migth be usefull.
